server/src/main/java/org/elasticsearch/search/aggregations/metrics/ScriptedMetricAggregationBuilder.java

```diff
@@ -198,6 +198,20 @@ protected ScriptedMetricAggregatorFactory doBuild(SearchContext context, Aggrega
 
         QueryShardContext queryShardContext = context.getQueryShardContext();
 
+        ScriptedMetricAggContexts.CombineScript.Factory compiledCombineScript;
+        Map<String, Object> combineScriptParams;
+        if (combineScript != null) {
+            compiledCombineScript = queryShardContext.getScriptService().compile(combineScript,
+                ScriptedMetricAggContexts.CombineScript.CONTEXT);
+            combineScriptParams = combineScript.getParams();
+        } else {
```

Comment:
I think we should leave lines 204-206 here where they were below but just have the `null` check here so here its just:
```
if (combineScript == null) {
    throw new IllegalArgumentException("[combineScript] must not be null: [" + name + "]");
}
```

And then below after the map script parts we have:
```
ScriptedMetricAggContexts.CombineScript.Factory compiledCombineScript = queryShardContext.getScriptService().compile(combineScript, ScriptedMetricAggContexts.CombineScript.CONTEXT);
Map<String, Object> combineScriptParams = combineScript.getParams();
```

Reply:
Making this change causes all the integ tests to fail and I haven't been able to debug why yet.

Reply:
all = all the scripted metric IT integ tests

Reply:
As I said, this combination fails the integration tests and I'm not sure why. It seems the return values of the test scripts are returning different values if done this way.

Reply:
So, I'm currently debugging issues with the dummy scripts interfering with return values. It takes a while since they are integ tests. Just letting you know I'm still working on this.